TitechAppLite/Decoder/IcsDecoder.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  IcsDecoder.swift
+//  TitechAppLite
+//
+//  Created by hoshiro ando on 2020/09/06.
+//  Copyright © 2020 hoshiro ando. All rights reserved.
+//
+
+import Foundation
+
+class IcsDecoder {
+    static func icsDecoder(icsString: String) -> [ICSEvent] {
+        let TZOFFSET = "TZOFFSETTO"
+        let BEGIN = "BEGIN:VEVENT"
+        let DTSTAMP = "DTSTAMP"
+        let DTEND = "DTEND"
+        let DTSTART = "DTSTART"
+        let LOCATION = "LOCATION"
+        let DESCRIPTION = "DESCRIPTION"
+        let SUMMARY = "SUMMARY"
+        let UID = "UID"
+        let END = "END:VEVENT"
+        var flag = false
+        var tzoffset = "0000"
+        var result: [ICSEvent] = []
+        
+        var tmpdtstamp: Date?
+        var tmpdtend: Date?
+        var tmpdtstart: Date?
+        var tmplocation: String?
+        var tmpdescription: String?
+        var tmpsummary: String?
+        var tmpuid: String?
+        
+        icsString.enumerateLines{ line, stop in
+            if line.hasPrefix(TZOFFSET) {
+                let l = line.replacingOccurrences(of: TZOFFSET + ":", with: "")
+                tzoffset = l
+            }
+            if line.hasPrefix(BEGIN) {
+                flag = true
+            }
+            
+            if flag {
+                if line.hasPrefix(DTSTAMP) {
+                    let arr = line.components(separatedBy: ":")
+                    if arr.count < 2 {
+                        return
+                    }
+                    tmpdtstamp = ISO8601StringtoDateTime(dateString: arr[1], tzoffset: tzoffset)
```

Comment:
> Resetting this property can incur a significant performance cost, as it may cause internal state to be regenerated.

https://developer.apple.com/documentation/foundation/iso8601dateformatter/1643324-formatoptions

とかあるので毎回 `ISO8601DateFormatter` を生成するのはやめた方が良さそう

Reply:
引数で`ISO8601DateFormatter`を受け取るようにしました